src/plugins/controls/public/options_list/components/options_list_popover_suggestions.tsx

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React, { useMemo } from 'react';
+
+import { EuiFilterSelectItem, EuiSpacer, EuiIcon } from '@elastic/eui';
+import { useReduxEmbeddableContext } from '@kbn/presentation-util-plugin/public';
+
+import { OptionsListReduxState } from '../types';
+import { OptionsListStrings } from './options_list_strings';
+import { optionsListReducers } from '../options_list_reducers';
+
+interface OptionsListPopoverSuggestionsProps {
+  showOnlySelected: boolean;
+}
+
+export const OptionsListPopoverSuggestions = ({
+  showOnlySelected,
+}: OptionsListPopoverSuggestionsProps) => {
+  // Redux embeddable container Context
+  const {
+    useEmbeddableDispatch,
+    useEmbeddableSelector: select,
+    actions: { replaceSelection, deselectOption, selectOption, selectExists },
+  } = useReduxEmbeddableContext<OptionsListReduxState, typeof optionsListReducers>();
+  const dispatch = useEmbeddableDispatch();
+
+  // Select current state from Redux using multiple selectors to avoid rerenders.
+  const existsSelectionInvalid = select((state) => state.componentState.existsSelectionInvalid);
+  const invalidSelections = select((state) => state.componentState.invalidSelections);
+  const availableOptions = select((state) => state.componentState.availableOptions);
+
+  const selectedOptions = select((state) => state.explicitInput.selectedOptions);
+  const existsSelected = select((state) => state.explicitInput.existsSelected);
+  const singleSelect = select((state) => state.explicitInput.singleSelect);
+  const hideExists = select((state) => state.explicitInput.hideExists);
+  const exclude = select((state) => state.explicitInput.exclude);
+
+  const loading = select((state) => state.output.loading);
+
+  // track selectedOptions and invalidSelections in sets for more efficient lookup
+  const selectedOptionsSet = useMemo(() => new Set<string>(selectedOptions), [selectedOptions]);
+  const invalidSelectionsSet = useMemo(
+    () => new Set<string>(invalidSelections),
+    [invalidSelections]
+  );
+  const suggestions = showOnlySelected ? selectedOptions : availableOptions;
+
+  if (
+    !loading &&
+    (!suggestions || suggestions.length === 0) &&
+    !(showOnlySelected && existsSelected)
+  ) {
```

Comment:
Note that, because of this logic, the `exists` query will not show up when searching regardless of if it's selected or not.

![Nov-02-2022 16-13-51](https://user-images.githubusercontent.com/8698078/199612660-f2ff5392-4e0d-46ee-9e70-39a845763c45.gif)

I did it this way because (a) this is the same way it works with any other selection (i.e. options that are selected don't show up if the search doesn't apply) and (b) because the `exists` query is always at the top, users shouldn't have to search for it. However, I'm not necessarily convinced about this behaviour, so interested in opinions! @andreadelrio

Reply:
I think this makes sense.